virtcontainers/qemu.go

```diff
@@ -837,6 +853,64 @@ func (q *qemu) hotplugRemoveCPUs(amount uint32) (uint32, error) {
 	return amount, q.sandbox.storage.storeHypervisorState(q.sandbox.id, q.state)
 }
 
+func (q *qemu) hotplugMemory(slot uint32, memMB uint64, op operation) error {
+	// We do not support memory hot unplug.
+	if op == removeDevice {
+		return errors.New("cannot hot unplug memory device")
+	}
+
+	// rule out extremely large parameter
+	if memMB > math.MaxInt32 {
+		return fmt.Errorf("cannot hotplug %d MB memory: size too large", memMB)
```

Comment:
Nit: The config file for kata ask for memory in MiB we may want to follo the same metric.